Products/CMFPlone/browser/search.py

```diff
@@ -189,7 +189,8 @@ def types_list(self):
         # only show those types that have any content
         catalog = getToolByName(self.context, "portal_catalog")
         used_types = catalog._catalog.getIndex("portal_type").uniqueValues()
-        return self.filter_types(list(used_types))
+        filtered_list= self.filter_types(list(used_types))
```

Comment:
> Talking of sorting: the standard sort() or sorted() method should not be applied because they do not deal correctly with unicode strings in the context of the configured locale using pyuca:
> https://github.com/jtauber/pyuca

please take the comment in the referenced issue into account ;)

Reply:
problem here is, that this only sorts the internal stored `portal_type` (eg. `Document`) but gets later translated in the template to `Seite` (german) ... this needs to be refactored to a translated and sorted vocabulary with internal type as key and translated `type_info.Title()` as label

Reply:
[pyuca](https://github.com/jtauber/pyuca) is 7 years old, use [pyICU](https://gitlab.pyicu.org/main/pyicu) instead (if possible).
```python
>>> from icu import RuleBasedCollator
>>> collator = RuleBasedCollator(
...     """
...     &A<ą<<<Ą
...     &C<ć<<<Ć
...     &E<ę<<<Ę
...     &L<ł<<<Ł
...     &N<ń<<<Ń
...     &O<ó<<<Ó
...     &S<ś<<<Ś
...     &Z<ź<<<Ź<ż<<<Ż
...     """
... )
>>> sorted(polish_names, key=collator.getSortKey)
['Ludmiła', 'Łukasz', 'Zbigniew', 'Żaneta']
```

You could also use locale() sorting capability as stated here: https://realpython.com/python-sort-unicode-strings/#leverage-pythons-locale-module-to-sort-unicode-strings
```python
>>> locale.setlocale(locale.LC_COLLATE, "pl_PL.UTF-8")
'pl_PL.UTF-8'
>>> sorted(polish_names, key=locale.strxfrm)
['Ludmiła', 'Łukasz', 'Zbigniew', 'Żaneta']
```
but beware of potential problems setting the locale, and also you need all the language locales set in your system.

Note: do this need a test too?

Reply:
I'd say: use the existing vocabulary for the searchfilter here and take care of sorting in `plone.app.vocabulary` -> new PR ...

Reply:
> The problem of pyICU is the scarse availability as compiled wheels in pypi, so we would depend on this. Windows has builds: https://github.com/cgohlke/pyicu-build

Attention with pyICU because it requires the related library to be installed on your OS. It's is no installed by default and would require an update of the Plone installation instructions for the related OS or distribution...also not sure about compatibility/availability of ICU on Windows.

Reply:
I'm stuck. I don't know how to make those changes. @petschki, @zopyx, and @yurj, can you please point me in the right direction?

Reply:
Point is that need one of the following solutions which basically can not directly integrated:
- `locale.setlocatel()` - an anti-pattern for switching the locale while an app is running
- `pyuca` - likely too old and unmaintained (but working for me under Python 3.12, but a new dependency)
- `pyICU`- primary choice but a new huge dependency that is likely unwanted as a dependency of Plone

Sorting the portal_types based on their translated title should be good enough in most case. I doubt that you need collation (as proposed using `pyuca`, `pyICU`) for the few portal type strings.